tensorflow/python/distribute/input_lib.py

```diff
@@ -376,6 +398,21 @@ def out_of_range_fn(worker_index, device):
 
     return values.regroup(replicas)
 
+class DistributedIteratorForReplicas(DistributedIterator):
+  """Input Iterator for a distributed dataset on replicas."""
+  def __init__(self, input_workers, iterators, strategy):
+    super(DistributedIteratorForReplicas, self).__init__(input_workers, iterators, strategy)
+
+  def get_next(self, name=None):
+    """Returns the next input from the iterator for all replicas."""
+    if not self._enable_get_next_as_optional:
```

Comment:
This seems to be assuming that the shapes will always be static. Why is that the case?

Reply:
ping